homeassistant/components/light/fibaro.py

```diff
@@ -0,0 +1,185 @@
+https://home-assistant.io/components/light.fibaro/
+"""
+
+# pylint: disable=R1715
+
+import logging
+import threading
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS, ATTR_BRIGHTNESS_PCT, ATTR_HS_COLOR, ATTR_WHITE_VALUE,
+    ENTITY_ID_FORMAT, SUPPORT_BRIGHTNESS, SUPPORT_COLOR, SUPPORT_WHITE_VALUE,
+    Light)
+import homeassistant.util.color as color_util
+from homeassistant.components.fibaro import (
+    FIBARO_CONTROLLER, FIBARO_DEVICES, FibaroDevice)
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['fibaro']
+
+
+def scaleto255(value):
+    """Scale the input value from 0-100 to 0-255."""
+    if value < 3:
+        value = 0
+    if value > 97:
+        value = 100
+    return max(0, min(255, ((value * 256.0) / 100.0)))
+
+
+def scaleto100(value):
+    """Scale the input value from 0-255 to 0-100."""
+    if value < 2:
+        value = 0
+    if value > 253:
+        value = 255
+    return max(0, min(100, ((value * 100.4) / 255.0)))
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Perform the setup for Fibaro controller devices."""
+    if discovery_info is None:
+        return
+
+    add_entities(
+        [FibaroLight(device, hass.data[FIBARO_CONTROLLER])
+         for device in hass.data[FIBARO_DEVICES]['light']], True)
+
+
+class FibaroLight(FibaroDevice, Light):
+    """Representation of a Fibaro Light, including dimmable."""
+
+    def __init__(self, fibaro_device, controller):
+        """Initialize the light."""
+        self._supported_flags = 0
+        self._last_brightness = 0
+        self._color = (0, 0)
+        self._brightness = None
+        self._white = 0
+
+        self._update_lock = threading.RLock()
+        if 'levelChange' in fibaro_device.interfaces:
+            self._supported_flags |= SUPPORT_BRIGHTNESS
+        if 'color' in fibaro_device.properties:
+            self._supported_flags |= SUPPORT_COLOR
+        if 'setW' in fibaro_device.actions:
+            self._supported_flags |= SUPPORT_WHITE_VALUE
+        super().__init__(fibaro_device, controller)
+        self.entity_id = ENTITY_ID_FORMAT.format(self.ha_id)
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def hs_color(self):
+        """Return the color of the light."""
+        return self._color
+
+    @property
+    def white_value(self):
+        """Return the white value of this light between 0..255."""
+        return self._white
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        return self._supported_flags
+
+    def turn_on(self, **kwargs):
+        """Turn the light on."""
+        with self._update_lock:
+            if self._supported_flags & SUPPORT_BRIGHTNESS:
+                target_brightness = None
+                if ATTR_BRIGHTNESS_PCT in kwargs:
```

Comment:
This never happens, the light component does the percent -> byte conversion for you. So only `ATTR_BRIGHTNESS` is passed to your light platform.